aptly/publisher/__init__.py

```diff
@@ -186,6 +186,91 @@ def _get_publish(self):
                 return publish
         raise NoSuchPublish("Publish %s does not exist" % self.name)
 
+    def save_publish(self, aptly_dir="/srv/aptly/"):
+        """
+        Serialize publish in YAML
+        """
+        name = self.name.replace('/', '-')
+        file_name = aptly_dir + name + "-" + time.strftime("%Y%m%d%H%M%S")
+        lg.info("Saving publish %s in %s" % (name, file_name))
+        with open(file_name, 'w') as save_file:
+            save_file.write("publish: %s\n" % self.name)
+            save_file.write("name: %s\n" % file_name)
+            save_file.write("components:\n")
+            for component, snapshots in self.components.items():
+                packages = self.get_packages(component)
+                save_file.write("  - component: %s\n" % component)
+                save_file.write("    snapshot: %s\n" % snapshots[0])
+                save_file.write("    packages:\n")
+                for package in packages:
+                    (arch, name, version, ref) = self.parse_package_ref(package)
+                    save_file.write("      - package: %s\n" % name)
+                    save_file.write("        version: %s\n" % version)
+                    save_file.write("        arch: %s\n" % arch)
+                    save_file.write("        ref: %s\n" % ref)
+
+    def restore_publish(self, config, components=[], recreate=False):
+        """
+        Restore publish from config file
+        """
+        try:
+            self.load()
+            self.save_publish()
+            publish = True
+        except NoSuchPublish:
+            publish = False
+
+        new_publish_snapshots = []
+        to_publish = []
+
+        for saved_component in config.get('components'):
+            component_name = saved_component.get('component')
```

Comment:
if there's no component name, then file is corrupted and you should probably fail here